js/decklist/decklist.js

```diff
@@ -303,6 +307,56 @@ function sortDecklist(deck, sortorder) {
     for (i = 0; i < deck.length; i++) { deck[i] = [ deck[i][2], deck[i][0] ] }
   }
 
+  else if ( sortorder == 'type' ) {
+
+    var type_to_cards = {}
+
+    for (i = 0; i < deck.length; i++) {
+
+      // We're going to search by lower case
+      var lcard = deck[i][0].toLowerCase();
+
+      // Grab the card's type
+      if (lcard in cards) { type = cards[ lcard ]['t']; }
+      else { type = 'z'; } // Unknown
+
+      // Create the type subarray
+      if ( !(type in type_to_cards ) ) { type_to_cards[type] = []; }
```

Comment:
@Nightfirecat 

For consistency with the rest of the file:

      if ( !(color in color_to_cards ) ) { color_to_cards[color] = []; }
      if ( !(cmc in cmc_to_cards ) ) { cmc_to_cards[cmc] = []; }
      if ( !(type in type_to_cards ) ) { type_to_cards[type] = []; }

Personally, this style makes my eyes bleed, but it's not my project, I'm just a guest here.